src/frequenz/client/assets/types.py

```diff
@@ -0,0 +1,174 @@
+# License: MIT
+# Copyright © 2025 Frequenz Energy-as-a-Service GmbH
+
+"""Type definitions for the Assets API client.
+
+This module contains the core data types used by the Assets API client,
+including data classes for representing assets, microgrids, and related entities.
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from typing import Any, Optional
+
+from frequenz.api.common.v1alpha8.grid.delivery_area_pb2 import (
+    DeliveryArea as PBDeliveryArea,
+)
+from frequenz.api.common.v1alpha8.microgrid.microgrid_pb2 import (
+    Microgrid as PBMicrogrid,
+)
+from frequenz.api.common.v1alpha8.types.location_pb2 import Location as PBLocation
+from frequenz.client.common.microgrid import EnterpriseId, MicrogridId
+
+
+@dataclass(frozen=True)
+class DeliveryArea:
+    """A wrapper class for the protobuf DeliveryArea message.
+
+    A delivery area is a geographical area that is served by a microgrid.
+    """
+
+    code: str
+    code_type: str
+
+    @staticmethod
+    def from_protobuf(pb: PBDeliveryArea) -> "DeliveryArea":
+        """
+        Create a DeliveryArea instance from a protobuf message.
+
+        Args:
+            pb: The protobuf DeliveryArea message.
+
+        Returns:
+            A new DeliveryArea instance populated with data from the protobuf message.
+        """
+        return DeliveryArea(code=pb.code, code_type=str(pb.code_type))
+
+    def to_dict(self) -> dict[str, Any]:
+        """
+        Convert the DeliveryArea instance to a dictionary.
+
+        Returns:
+            A dictionary containing the delivery area details.
+        """
+        return {"code": self.code, "code_type": self.code_type}
```

Comment:
These `to_dict()` are not necessary, you can use [`dataclasses.asdict()`](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict) on any `dataclass`.